### PR TITLE
More sophisticated data-outage logic

### PR DESCRIPTION
**Pull request by phildarnowsky**
Here we update the data-status logic to display the black outage banner in case:

* There are at least 10 vehicles currently in revenue service; and,
* At least 20% of vehicles are either missing data from either source, or have data over 150 seconds old from either source.

### CODE REVIEW

File: test/concentrate/vehicle_position_test.exs [hunk 1]
```diff
@@ -319,6 +328,42 @@ defmodule Concentrate.VehiclePositionTest do
 
       assert Mergeable.merge(first, second) == expected
     end
+
+    test "merge/2 retains all timestamps" do
```
Comment (lemald) [inline test/concentrate/vehicle_position_test.exs:332]: Hmm, am I misunderstanding the purpose of this test, or should there be some actual values in `last_updated_by_source`?

Comment (phildarnowsky) [reply test/concentrate/vehicle_position_test.exs:332]: you understand correctly...I will look into this.

**Comment by phildarnowsky** [comment]
@lemald ready for another look

**Review by lemald — APPROVED** [review]
Looks good to me! 👍

**Pull request merged**